### ISSUE TITLE
Allow @key to reference non-required fields

### ISSUE THREAD

**Issue by wizage** [aws-amplify/amplify-cli#3686]
**Describe the bug**
Today @key doesn't allow you to reference non-mandatory fields. This can hinder other transforms and custom revolvers that might take a null field and fill in a default if no value is given.

**Amplify CLI Version**
4.16.1

**To Reproduce**
An example:

```
type Answer @model @key(fields: ["owner", "gameID"])
@auth(
    rules: [
        {allow: owner, ownerField: "owner"}
    ]
)
{
    owner: String
    gameID: ID!
    answer: [Int]
}
```
This example, fails to compile today but should compile as the resolver will auto fill the owner field in if no value is provide due to the auth resolver.

**Expected behavior**
I expect a flag to be able to set in the @key transform that will allow me to disable the required non-nullable check. Or this should not stop the compile and should throw a warning letting me know that there could be possible issues down the road but will allow me to still push. 

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version: v12.13.1

**Comment by RossWilliams** [comment]
See duplicate #3275

**Comment by wizage** [comment]
> See duplicate #3275

I would disagree but it does look like it does relate/might be a duplicate of #1657
